docs/react/react-native.md

```diff
@@ -76,3 +76,52 @@ export function useRefreshOnFocus<T>(refetch: () => Promise<T>) {
 ```
 
 In the above code, `refetch` is skipped the first time because `useFocusEffect` calls our callback on mount in addition to screen focus.
+
+## Disable re-renders on out of focus Screens
+
+In some situations, including performance concerns, you may want to stop re-renders when a React Native screen gets out of focus. To achieve this we can use `useFocusEffect` from `@react-navigation/native` together with the `notifyOnChangeProps` query option.
+
+This custom hook provides a `notifyOnChangeProps` option that will return an empty array whenever a screen goes out of focus - effectively stopping any re-renders on that scenario. Whenever the screens gets in focus again, the behavior goes back to normal.
+
+```tsx
+import React from 'react'
+import { UseQueryOptions } from '@tanstack/react-query'
+import { useFocusEffect } from '@react-navigation/native'
+
+export function useFocusNotifyOnChangeProps(
+  notifyOnChangeProps: UseQueryOptions['notifyOnChangeProps']
```

Comment:
What is the best way to get the `notifyOnChangeProps` type?

Reply:
yeah that's the only way right now, unless you want to extract and export the type separately?

https://github.com/TanStack/query/blob/dcda9200c9d29ae0b56b1d78ded0d9d721ba6b77/packages/query-core/src/types.ts#L266